src/graphics/Screen.cpp

```diff
@@ -2143,6 +2145,13 @@ void Screen::setFrames()
                     // just changed)
 
     setFastFramerate(); // Draw ASAP
+
+    if (currentFrameNum > numframes - 1) { // If we were on a frame that no longer exists
+        ui->switchToFrame(numframes - 2); // Attempt to return to last frame
+        } 
```

Comment:
What if that one doesn't exist as well? i'd rather return to the first frame, there's always one frame.

Reply:
Ah I think I might have caused some of this confusion. I made a (probably unclear) suggestion that it would be good to store `numFrames` at the end of the method, so that next time `setFrames()` is called, we can detect whether we were previously on the log buffer / settings frames, and make sure we return to them, even if the number of frames has now changed.